### PR TITLE
The first oncreate () when using glide gif to add. #issue 7

### PR DESCRIPTION
**Pull request by saechimdaeki**
Codebeat did not work after modification and opened a new PR. #7 

The first oncreate () when using glide gif to add. 
like this↓ 
 
exefff

### CODE REVIEW

**Comment by saechimdaeki** [comment]
I closed #8 and opened this PR.

**Comment by saechimdaeki** [comment]
Why is the codebeat still pending? Does the reviewer have to review before it starts?

**Comment by YassinAJDI** [comment]
@saechimdaeki Thanks for the PR! I like the idea, but I think it needs some improvements. For example, the loading gif could be fullscreen just like a splash screen. the Bottom menu and toolbar should be hidden;

**Comment by saechimdaeki** [comment]
@YassinAJDI   As you said, I made loading_Activity so that it can be played only in the first oncreate(). I gave 8000 mills to loading_activity.

**Comment by saechimdaeki** [comment]
@YassinAJDI First of all, is this enough for this pullrequest?

File: app/src/main/java/com/ajdi/yassin/popularmovies/ui/movieslist/MoviesActivity.java [hunk 1]
```diff
@@ -1,32 +1,29 @@
...
-
 public class MoviesActivity extends AppCompatActivity {
-
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
-
         if (savedInstanceState == null) {
+            Intent intent = new Intent(this, LoadingActivity.class);
```
Comment (saechimdaeki) [inline app/src/main/java/com/ajdi/yassin/popularmovies/ui/movieslist/MoviesActivity.java:20]: The reason why this function is added to this is because you can skip by pressing the Cancel button if the loading screen feels like a long time.

**Comment by saechimdaeki** [comment]
@YassinAJDI  Can you review it? If there is anything to improve, I want to do more.

File: .idea/misc.xml [hunk 2]
```diff
@@ -39,7 +39,7 @@
       </value>
     </option>
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8 (1)" project-jdk-type="JavaSDK">
```
Comment (YassinAJDI) [inline .idea/misc.xml]: why was this changed?

File: app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+import com.bumptech.glide.Glide;
+
+public class LoadingActivity extends Activity {
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_loading);
+        ImageView imageView=(ImageView)findViewById(R.id.image_loading);
+        Glide.with(getBaseContext()).load(R.raw.marvle).into(imageView);
```
Comment (YassinAJDI) [inline app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java]: Is the marvel gif free and open source?

File: app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        startLoading();
+    }
+    private void startLoading() {
+        Handler handler = new Handler();
+        handler.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                finish();
+            }
+        }, 8000);
```
Comment (YassinAJDI) [inline app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java]: 8 seconds is too long. can you reduce it something like 4s?

File: app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+
+import com.bumptech.glide.Glide;
+
+public class LoadingActivity extends Activity {
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_loading);
+        ImageView imageView=(ImageView)findViewById(R.id.image_loading);
```
Comment (YassinAJDI) [inline app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java]: You can remove the casting here

File: app/src/main/res/layout/activity_loading.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:orientation="vertical" android:layout_width="match_parent"
+    android:layout_height="match_parent">
```
Comment (YassinAJDI) [inline app/src/main/res/layout/activity_loading.xml:5]: on high-resolution devices, the gif looks small. any way to make the gif hight bigger or change the aspect ratio?

**Review by YassinAJDI — CHANGES_REQUESTED** [review]
Looks good. just a few changes before the merge.

File: app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java (hunk 3 above)
Comment (saechimdaeki) [reply app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java]: Marvel is said to be fine when the logo comes out. So I'll change it to another gif.

File: app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java (hunk 4 above)
Comment (saechimdaeki) [reply app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java]: I'll bring a gif that takes a total of 5 seconds and play it for 4 seconds.

File: app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java (hunk 5 above)
Comment (saechimdaeki) [reply app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java]: ok i will remove that

File: app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java (hunk 3 above)
Comment (YassinAJDI) [reply app/src/main/java/com/ajdi/yassin/popularmovies/LoadingActivity.java]: if it's open-source then no need to change it.

File: .idea/misc.xml (hunk 2 above)
Comment (saechimdaeki) [reply .idea/misc.xml]: I think this is because I don't have a .gitignore. Maybe it will go back to the way it was run on your computer.

File: app/src/main/res/layout/activity_loading.xml (hunk 6 above)
Comment (saechimdaeki) [reply app/src/main/res/layout/activity_loading.xml:5]: i will try this

**Comment by saechimdaeki** [comment]
@YassinAJDI  
I solved the resolution problem and I did so because 4.5 seconds seems more stable than 4 seconds.

marvel2

**Comment by saechimdaeki** [comment]
The reason for changing the gif was to meet the limit of 4 seconds, and the previous gif was that the pixel break occurred when the image size was increased, causing eye pain.

**Comment by YassinAJDI** [comment]
@saechimdaeki thanks for the update. However, I think the old gif looks better because of the Marvel logo is misleading.

**Comment by saechimdaeki** [comment]
@YassinAJDI  Done! 

marvel

File: .idea/misc.xml (hunk 2 above)
Comment (YassinAJDI) [inline .idea/misc.xml]: Can you accept this so I can merge

```suggestion
  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```

Comment (saechimdaeki) [reply .idea/misc.xml]: I didn't know the reason for the change, so I went back to the way

Comment (saechimdaeki) [reply .idea/misc.xml]: @YassinAJDI  finally done

**Comment by saechimdaeki** [comment]
@YassinAJDI  .idea/misc.xml   It seems that android Studio handled it because you and I are different versions of java. I understand that the change in this is not problematic. Still, it has been returned to its original condition.

File: .idea/misc.xml (hunk 2 above)
Comment (saechimdaeki) [reply .idea/misc.xml]: I've brought this back to its original state

**Review by YassinAJDI — APPROVED** [review]

**Comment by YassinAJDI** [comment]
@saechimdaeki Thanks, LGTM! This PR was merged 🎉

**Comment by saechimdaeki** [comment]
@YassinAJDI  Thank you my friend! I will follow you!

**Comment by YassinAJDI** [comment]
@saechimdaeki I followed you back :D
